### ISSUE TITLE
[Bug]: Single example not supported on arrays

### ISSUE THREAD

**Issue by Arithmomaniac** [microsoft/typespec#4546]
### Describe the bug

In [OpenAPI spec](https://swagger.io/docs/specification/v3_0/adding-examples/#array-example), when providing an example of `T[]` you can specify _either_ an single example `T`, _or_ multiple `T`s via a single `T[]`.

This is not how TypeSpec works today.

I would think fixing this would be easy enough to implement for all emitters; if the target does not support a `T` example for `T[]`, just turn it into a singleton array.

_From Sefaria's existing examples._

### Reproduction

[Playground](https://typespec.io/playground?c=QGV4YW1wbGUoI1sxXSkKbW9kZWwgYXJyYXlPZkludDEgaXMgaW50MzJbXTsKCskuMtIrMswr&e=%40typespec%2Fopenapi3&options=%7B%7D)

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that request the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/Microsoft/typespec/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by timotheeguerin** [comment]
I think this is not really specific to example this one but more of the lack of being able to decorate array element type.

The workaround for that is to create a reusable scalar for it.

**Comment by Arithmomaniac** [comment]
I'm not sure what you mean. If you look at the playground example, `arrayOfInt1` emits the OpenAPI one would expect once `arrayOfInt2` is removed.

**Comment by timotheeguerin** [comment]
i'm saying it is expected that it errors in the `arrayOfInt2` `@example` decorator.

it is still annotating the array and not its element type. every other decorator behave the same.

[Playground workaround example](https://typespec.io/playground?c=QGV4YW1wbGUoI1sxXSkKbW9kZWwgYXJyYXlPZkludDEgaXMgaW50MzJbXTsK0R8yxB9vdGhlcsUfyU0yKQpzY2FsYXLGHSBleHRlbmTHSjs%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)

**Comment by Arithmomaniac** [comment]
This is the responsible code, right?

https://github.com/microsoft/typespec/blob/3804d7a5881d86c8e064a41f7eb6b11a070c92eb/packages/compiler/src/lib/decorators.ts#L1559-L1575

What I'm hearing you say is that it's not practical to treat `target` as an array with typed information. But what about having a fallback (when `target` is an array) to test against `#[value]` if the test against `value` fails?

**Comment by timotheeguerin** [comment]
Yes that is the code, validation is only one part of it though, every emitter would also need to be updated to make sure they don't use the example meant for the element type on the array itself.

My main point here is not that it's not possible to do for example but its just against the general behavior of decorators and support for a single one would be confusing IMO. There is other decorators where it wouldn't be possible to have a fallback behavior and make it impossible to decorate the element type. Even in this case, suppose you have this type

```ts
@example([1, 2, 3]) 
model MyArray is unknown[];
```
how would I say i want this example to be about the `unknown` type vs the array itself so then you would still need a way to explicitly target the element type in some way.

I am also not sure of the real benefit of being able to do that. Either the element type is complex enough to be its own type where you can annotate it or showing examples of the array seems more relevant to me. I think it is fine keeping this issue open to see if there is more demand for this feature(or other decorators that want this) and if there is we can design a solution.

**Comment by Arithmomaniac** [comment]
Fair. It also would be leaning specifically into an OpenAPI-specific behavior that isn't really necessary for anything you'd want to do with OpenAPI anyway.

I'd just keep in mind whenever the example importer is written for tsp-openapi3 to "wrap" these sort of singleton examples into arrays if needed.

**Comment by markcowl** [comment]
@markcowl  add label

@mario-guerra  need a process for community feedback on new features

**Comment by markcowl** [comment]
@timotheeguerin  Please draft a more specific issue about the ability to target Array Items and then reference that issue when closing this one

**Comment by timotheeguerin** [comment]
Closing in favor of more general issue #7698
